src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs

```diff
@@ -1509,8 +1509,16 @@ async ValueTask WaitingSide(QuicStream stream, Task<long?> task, long errorCode)
                 {
                     // Writes must be closed, but whether successfully or not depends on the timing.
                     // Peer might have aborted reading side before receiving all the data.
-                    // Manicka TODO await writes closed
-                    Assert.True(stream.WritesClosed.IsCompleted);
+                    try
+                    {
+                        await stream.WritesClosed.WaitAsync(TimeSpan.FromSeconds(5));
```

Comment:
should the await always throw in this case? Or is it nondeterministic?

Reply:
Non-deterministics as the comment above explains. It depends on timing of the peer, dispose might be called before all the data arrived ==> abort, or after ==> success. This is not just because we buffer the data, MsQuic has it's own layer of buffering that behaves the same way.